src/destructuring.rs

```diff
@@ -12,14 +12,24 @@ use crate::{
     types::{TypeF, Types},
 };
 
+#[derive(Debug, PartialEq, Clone)]
+pub enum MatchAssignment {
+    /// An assignment match like `{ ..., a = b, ... }`
+    Ident(Ident),
+    /// A nested record pattern like `{ ..., a = { b, c }, ... }`
+    RecordPattern(RecordPattern),
+    /// An identified nested record pattern like `{ ..., a = b @ { c, d }, ... }`
+    IdentPattern(Ident, RecordPattern),
```

Comment:
IMHO, the name `IdentPattern` is slighty confusing/inconsistent with the previous variants (it looks like it's a pattern destructuring an ident). In the end it's still a record pattern, but with something additional. I don't know the general name of such a top-level capture on the top of my head, unfortunately. Still, for the record, the OCaml manual refers to them as "alias patterns". Maybe `AliasRecordPattern`? `AliasedRecordPattern`?

Reply:
I'd originally gone with `IdentifiedPattern` but felt it was a little too long. I like `AliasedRecordPattern` a lot, so let's go with that. :+1: